### DISCUSSION TITLE
JDBC Connection Inconsistency

### DISCUSSION THREAD

**Discussion by ryanhamilton** [duckdb/duckdb#15029] [Ideas]
JDBC URLs tend to work as described here: https://github.com/duckdb/duckdb/discussions/10007#discussioncomment-7869448

> "DuckDB internally manages a cache of open connections, and if you open a new connection with the same configuration as an existing open connection, it'll return a new connection to the same database"

i.e. connections to "jdbc:duckdb:a" AND "jdbc:duckdb:a" return the same database.

1. This doesn't seem to be true for in-memory "jdbc:duckdb:" seems to return different databases per connection held, can someone confirm this is intentional? 

2. If it's possible it would be really nice to allow "jdbc:duckdb:memory:a" to always return the same database but for "jdbc:duckdb:memory:b" to return a different in-memory database. Currently I don't see a way to have multiple separate in-memory duckdb databases?

To give credit where it's due, I've seen the idea in H2 database which has always had this behaviour: http://www.h2database.com/html/features.html#in_memory_databases  

> Sometimes multiple connections to the same in-memory database are required. In this case, the database URL must include a name. Example: jdbc:h2:mem:db1. Accessing the same database using this URL only works within the same virtual machine and class loader environment.

**Comment by Tishj** [comment]
I'm pretty sure that exact syntax enables this, if it doesnt that might be a bug in duckdb-jdbc
